pkg/antctl/raw/packetcapture/command.go

```diff
@@ -0,0 +1,337 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packetcapture
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/ptr"
+
+	"antrea.io/antrea/pkg/antctl/raw"
+	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	antrea "antrea.io/antrea/pkg/client/clientset/versioned"
+	"antrea.io/antrea/pkg/util/env"
+)
+
+var (
+	defaultTimeout          = time.Second * 60
+	maxPacketCaptureTimeout = time.Second * 300
+	Command                 *cobra.Command
+	getClients              = getConfigAndClients
+	getCopier               = getPodFileCopier
+	defaultFS               = afero.NewOsFs()
+)
+
+type options struct {
+	source    string
+	dest      string
+	nowait    bool
+	timeout   time.Duration
+	number    int32
+	flow      string
+	outputDir string
+}
+
+var option = &options{}
+
+var packetCaptureExample = `  Start capturing packets from pod1 to pod2, both Pods are in Namespace default
+  $ antctl packetcaputre -S pod1 -D pod2
+  Start capturing packets from pod1 in Namespace ns1 to a destination IP
+  $ antctl packetcapture -S ns1/pod1 -D 192.168.123.123
+  Start capturing UDP packets from pod1 to pod2, with destination port 1234
+  $ antctl packetcapture -S pod1 -D pod2 -f udp,udp_dst=1234
+  Save the packets file to a specified directory
+  $ antctl packetcapture -S 192.168.123.123 -D pod2 -f tcp,tcp_dst=80 -o /tmp
+`
+
+func init() {
+	Command = &cobra.Command{
+		Use:     "packetcapture",
+		Short:   "Start capture packets",
+		Long:    "Start capturing packets on the target flow",
+		Aliases: []string{"pc", "packetcaptures"},
+		Example: packetCaptureExample,
+		RunE:    packetCaptureRunE,
+	}
+
+	Command.Flags().StringVarP(&option.source, "source", "S", "", "source of the the PacketCapture: Namespace/Pod, Pod, or IP")
+	Command.Flags().StringVarP(&option.dest, "destination", "D", "", "destination of the PacketCapture: Namespace/Pod, Pod, or IP")
+	Command.Flags().Int32VarP(&option.number, "number", "n", 0, "target number of packets to capture, the capture will stop when it is reached")
+	Command.Flags().StringVarP(&option.flow, "flow", "f", "", "specify the flow (packet headers) of the PacketCapture, including tcp_src, tcp_dst, udp_src, udp_dst")
+	Command.Flags().BoolVarP(&option.nowait, "nowait", "", false, "if set, command returns without retrieving results")
+	Command.Flags().StringVarP(&option.outputDir, "output-dir", "o", ".", "save the packets file to the target directory")
+}
+
+var protocols = map[string]int32{
+	"icmp": 1,
+	"tcp":  6,
+	"udp":  17,
+}
+
+func getPodFileCopier(cmd *cobra.Command) (raw.PodFileCopier, error) {
```

Comment:
instead of having `cmd *cobra.Command` as the function parameter, I would just use (`config *rest.Config, client kubernetes.Interface)`

This way we don't need to return an error in this function, we don't call `getClients` twice (even though the cost of doing that doesn't matter), and we don't need to ignore the error returned by `getCopier` below (which can create confusion unless you know what `getCopier` is actually doing).